### PR TITLE
feat(DropdownV2): allow components as items

### PR DESCRIPTION
**Pull request by justinpurcell1**
Closes IBM/carbon-components-react#1495

Modify DropdownV2 to render items as components instead of strings if desired. Note that the original issue specified DropdownItem, but the change has been applied to DropdownV2 since the old Dropdown is deprecated.

#### Changelog

**Changed**

* DropdownV2 now accepts an `itemToElement` prop and prefers that over `itemToString`

### CODE REVIEW

File: src/components/DropdownV2/DropdownV2-story.js [hunk 1]
```diff
@@ -88,7 +116,8 @@ storiesOf('DropdownV2', module)
       `,
     })(() => (
       <div style={{ width: 300 }}>
-        <DropdownSkeleton />&nbsp;
+        <DropdownSkeleton />
+        &nbsp;
```
Comment (justinpurcell1) [inline src/components/DropdownV2/DropdownV2-story.js]: This looks like a side effect of prettier. The commit hooks also changed the bullets in .md files and added various newlines to .js files.

Comment (joshblack) [reply src/components/DropdownV2/DropdownV2-story.js]: Hi @justinpurcell1! Most likely this is due to using something other than `yarn` to manage dependencies (e.g. prettier version is different locally than what the project uses).

**Comment by joshblack** [comment]
Hi @justinpurcell1! 👋 Thanks so much for taking the time to make this PR 😄 

`DropdownV2` is a tricky one. The requirements moving from Dropdown, DropdownItem towards the new API may be reverted as we're noting more and more valid use-cases (like this one) coming up as folks are transitioning. We have this open issue to potentially handle both signatures: https://github.com/IBM/carbon-components-react/issues/1364

Might be worthwhile having this conversation over under that to see if this could help out a lot of these use-cases that people are noting 👍

**Comment by justinpurcell1** [comment]
@joshblack this can serve as a jumping off point from which to modify either DropdownV2 or DropdownItem. With this change it has most of the functionality but has some issues due to the click handling in DropdownItem

**Comment by joshblack** [comment]
> @joshblack this can serve as a jumping off point from which to modify either DropdownV2 or DropdownItem.

As a general policy, we try and have conversations in issues before Pull Requests (if applicable) so it would be great to start off over in that issue 👍

**Comment by justinpurcell1** [comment]
@jnm2377 yeah, that style issue shouldn't be too bad to change in a real implementation. My biggest concern was the difference in click handling and selection between Dropdown and DropdownV2

**Comment by justinpurcell1** [comment]
@jnm2377 is there anything else I should clean up? Are there any more steps required to get this merged in?

**Comment by jnm2377** [comment]
No, everything looks good. My apology for the delays. Thanks for the contribution @justinpurcell1

File: src/components/DropdownV2/DropdownV2.js [hunk 2]
```diff
@@ -141,7 +154,7 @@ export default class DropdownV2 extends React.Component {
                     isActive={selectedItem === item}
                     isHighlighted={highlightedIndex === index}
                     {...getItemProps({ item, index })}>
-                    {itemToString(item)}
+                    <ItemToElement key={itemToString(item)} {...item} />
```
Comment (asudoh) [inline src/components/DropdownV2/DropdownV2.js]: HI @justinpurcell1 👋 when `item` is a string, the spread operator here seems to make it an array of characters (ref: https://github.com/IBM/carbon-components-react/issues/1586). Do you want to use this render-element approach here only if `itemToElement` is given?

**Pull request merged**